### PR TITLE
Wait for services to start on windows

### PR DESCRIPTION
**Pull request by a-crate**
Fixes sql server CI failures

Passed locally.

/cc @zmarano @koln67 @drewhli

### CODE REVIEW

File: imagetest/test_suites/packagevalidation/windows_version_test.go [hunk 1]
```diff
@@ -235,23 +236,37 @@ func TestServicesState(t *testing.T) {
...
-		status := strings.TrimSpace(out.Stdout)
-		out, err = utils.RunPowershellCmd(fmt.Sprintf("(Get-Service -Name %s -ErrorAction Stop).StartType", service))
-		if err != nil {
-			t.Fatalf("Error getting service starttype: %v %v", err, out.Stderr)
+	checkServices := func() []error {
+		var errs []error
+		for _, service := range services {
+			out, err := utils.RunPowershellCmd(fmt.Sprintf("(Get-Service -Name %s -ErrorAction Stop).Status", service))
+			if err != nil {
+				errs = append(errs, fmt.Errorf("Error getting service status: %v %v", err, out.Stderr))
```
Comment (drewhli) [inline imagetest/test_suites/packagevalidation/windows_version_test.go:244]: Nit: include `service` name in error message.

File: imagetest/test_suites/packagevalidation/windows_version_test.go [hunk 2]
```diff
@@ -235,23 +236,37 @@ func TestServicesState(t *testing.T) {
...
+				return errs
+			}
+			status := strings.TrimSpace(out.Stdout)
+			out, err = utils.RunPowershellCmd(fmt.Sprintf("(Get-Service -Name %s -ErrorAction Stop).StartType", service))
+			if err != nil {
+				errs = append(errs, fmt.Errorf("Error getting service starttype: %v %v", err, out.Stderr))
```
Comment (drewhli) [inline imagetest/test_suites/packagevalidation/windows_version_test.go:250]: Ditto above

File: imagetest/test_suites/packagevalidation/windows_version_test.go [hunk 3]
```diff
@@ -235,23 +236,37 @@ func TestServicesState(t *testing.T) {
...
+	}
+	for {
+		errs := checkServices()
+		if len(errs) == 0 {
+			break
 		}
-		if startType != "Automatic" {
-			t.Errorf("service %s has unexpected StartType, got %s want Automatic", service, startType)
+		if err := ctx.Err(); err != nil {
+			t.Fatalf("context expired before services reached expected state: %v", err)
```
Comment (drewhli) [inline imagetest/test_suites/packagevalidation/windows_version_test.go:269]: I wonder if it'll be better to format the `err` part of this message ourselves: something like including linebreaks between each error message in the array for better readability.

Comment (a-crate) [reply imagetest/test_suites/packagevalidation/windows_version_test.go]: Updated error message with last state mismatch details, one line each

File: imagetest/test_suites/packagevalidation/windows_version_test.go (hunk 2 above)
Comment (a-crate) [reply imagetest/test_suites/packagevalidation/windows_version_test.go]: Done.

File: imagetest/test_suites/packagevalidation/windows_version_test.go (hunk 1 above)
Comment (a-crate) [reply imagetest/test_suites/packagevalidation/windows_version_test.go]: Done.

**Review by drewhli — APPROVED** [review]

**Pull request merged**